### DISCUSSION TITLE
Point Stat: MET/11.1.0 and METplus/5.1.0 could not find matched pairs (Python Embedding for Point Observations)

### DISCUSSION THREAD

**Discussion by pratiman-91** [dtcenter/METplus#2685] [Configuration] [answered]
Hi,
I am running `PointStat` analysis using MET/METplus. I am using `PYTHON_XARRAY` to read forecast files (UM files) and `PYTHON_NUMPY` to read point observations. I got zero matched pairs, if there are multiple valid time (extra +/- 24 hours). However, when I provide with exact valid time it works. 

Is this behavior expected?  

## Working with exact valid time
```
DEBUG 2: Processing t2/Surface versus dbt/L0, for observation type ADPSFC, over region FULL, for interpolation method BILIN(4), using 7 matched pairs.
DEBUG 3: Number of matched pairs   = 7
DEBUG 3: Observations processed    = 7
DEBUG 3: Rejected: station id      = 0
DEBUG 3: Rejected: obs var name    = 0
DEBUG 3: Rejected: valid time      = 0
DEBUG 3: Rejected: bad obs value   = 0
DEBUG 3: Rejected: off the grid    = 0
DEBUG 3: Rejected: topography      = 0
DEBUG 3: Rejected: level mismatch  = 0
DEBUG 3: Rejected: quality marker  = 0
DEBUG 3: Rejected: message type    = 0
DEBUG 3: Rejected: masking region  = 0
DEBUG 3: Rejected: bad fcst value  = 0
DEBUG 3: Rejected: bad climo mean  = 0
DEBUG 3: Rejected: bad climo stdev = 0
DEBUG 3: Rejected: mpr filter      = 0
DEBUG 3: Rejected: duplicates      = 0
DEBUG 2: Computing Categorical Statistics.
DEBUG 2: Computing Scalar Partial Sums and Continuous Statistics.
```
## Not working with multiple valid times
```
DEBUG 2: Processing t2/Surface versus dbt/L0, for observation type ADPSFC, over region FULL, for interpolation method BILIN(4), using 0 matched pairs.
DEBUG 2: Number of matched pairs   = 0
DEBUG 2: Observations processed    = 549
DEBUG 2: Rejected: station id      = 0
DEBUG 2: Rejected: obs var name    = 0
DEBUG 2: Rejected: valid time      = 549
DEBUG 2: Rejected: bad obs value   = 0
DEBUG 2: Rejected: off the grid    = 0
DEBUG 2: Rejected: topography      = 0
DEBUG 2: Rejected: level mismatch  = 0
DEBUG 2: Rejected: quality marker  = 0
DEBUG 2: Rejected: message type    = 0
DEBUG 2: Rejected: masking region  = 0
DEBUG 2: Rejected: bad fcst value  = 0
DEBUG 2: Rejected: bad climo mean  = 0
DEBUG 2: Rejected: bad climo stdev = 0
DEBUG 2: Rejected: mpr filter      = 0
DEBUG 2: Rejected: duplicates      = 0
```

Please let me know if you need more details.
Thanks!

**Comment by DanielAdriaansen** [comment]
Greetings @pratiman-91!

From the log output that you shared, I am able to determine that for one run you get **7 matched pairs**, which you say is from using the "exact valid time". Then, for another run, you get **549 rejections** due to valid time, when you use a valid time window.

I am slightly confused why there are not 549-7 (542 rejections for valid time) in the first case (using exact valid time).

I think in order to best assist you, we'd need some more information such as:
1. Your METplus wrappers configuration file
2. Information about the point observations you are passing to `PointStat` from Python. Ideally this would be your Python script and sample data that your Python script reads, but it could also be output from a print statement in your Python script showing what the data looks like.
3. Information about the forecast data you are passing to `PointStat` from Python. At a minimum, the `attrs` that you are setting in your Python script, but ideally the Python script and a sample data that Python can read.
4. Your complete log file from running METplus Wrappers

Any of those additional items you can provide will greatly assist us. Thanks!

**Reply by DanielAdriaansen** [reply]
I will also note that there was a bug in `MET-11.1.0` with Python embedding for point observations for the `valid_time`. It was fixed and has been patched in a bugfix release of `MET-11.1.1`. I can't say for sure that this bug is affecting your run, but since you are using the capability that the bug affected (Python embedding for point observations), I would strongly recommend upgrading to `MET-11.1.1+` before continuing, and trying that to see if it resolves the issue you describe.

More details about the fix and bug can be found here:
https://github.com/dtcenter/MET/pull/2899

**Reply by pratiman-91** [reply, marked as answer]
Thank you for the reply. I will check with the newer version and get back to you. From the bug thread, it seems like I am facing the same problem. 

Update: It is working with MET-11.1.1. Thanks for the support!

**Reply by pratiman-91** [reply]
Hi @DanielAdriaansen,
It is working with `MET-11.1.1`. Thanks for the support!
